docs: Notification doc zapier [WEB-216] (#5305)
determined-ai/determined: docs/integrations/notification/zapier.rst

################
 Through Zapier
################

This section will walk through the steps needed to set up Zapier webhook to to receive updates from
Determined.

> The steps for set up Zapier webhook are:

Reviewer: ```suggestion
The steps to set up Zapier webhook are:
```

   :width: 100%

Construct input data as following:

-  webhook_signing_key: match the ``webhook_signing_key`` in Determined.
-  timestamp: ``X-Determined-AI-Signature-Timestamp`` from request header.
-  signature: ``X-Determined-AI-Signature`` from request header.
-  payload: raw body of request.

.. image:: /assets/images/zapier_code_input.png
   :width: 100%

Input code as following:

.. code::

   import hmac, hashlib, json

   signing_key = input_data['webhook_signing_key']
   timestamp = input_data['timestamp']
   signature = input_data['signature']
   payload = json.loads(input_data['payload'])

   calculated_signature = hmac.new(signing_key.encode(), f"{timestamp},{payload}".encode(), digestmod=hashlib.sha256).hexdigest()

   if calculated_signature == signature:
       return {"result": "PASS", "payload": payload}
   return {"result": "Signature cannot be verified, request might not be legit"}

Under **Test Action**, test the code above, you should be able to see that verification has passed.

.. image:: /assets/images/zapier_code_result.png
   :width: 100%

.. _configuring_destination:

*************************
 Configuring Destination
*************************

> Finally, you can configure where to proceed under each senerio by adding more actions. For example,

Reviewer: ```suggestion
Finally, you can configure where to proceed under each scenario by adding more actions. For example,
```

> send out alert when verification fail, or send out email with experiment information when

Reviewer: ```suggestion
send out an alert when verification fails, or send out an email with experiment information when
```
